src/Utilities/Compiler/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -287,6 +287,57 @@ public static Accessibility DetermineMinimalAccessibility(this ITypeSymbol typeS
             return typeSymbol.Accept(MinimalAccessibilityVisitor.Instance);
         }
 
+        public static bool HasAnyCollectionCountProperty(this ITypeSymbol invocationTarget, WellKnownTypeProvider wellKnownTypeProvider)
+        {
+            const string countPropertyName = "Count";
```

Comment:
I kept it as a local constant for now but it might be better to create some kind of `WellKnownMethodNames` like we have `WellKnownTypeNames` to avoid string duplication.

Reply:
I think this is fine.